Blockzilla/Modules/WebView/WebViewController.swift

```diff
@@ -0,0 +1,146 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import UIKit
+import WebKit
+
+protocol BrowserState {
+    var url: URL? { get }
+    var isLoading: Bool { get }
+    var canGoBack: Bool { get }
+    var canGoForward: Bool { get }
+    var estimatedProgress: Double { get }
+}
+
+protocol WebController {
+    weak var delegate: WebControllerDelegate? { get set }
+    var canGoBack: Bool { get }
+    var canGoForward: Bool { get }
+
+    func load(_ request: URLRequest)
+}
+
+protocol WebControllerDelegate: class {
+    func webControllerDidStartNavigation(_ controller: WebController)
+    func webControllerDidFinishNavigation(_ controller: WebController)
+    func webController(_ controller: WebController, didFailNavigationWithError error: Error)
+    func webController(_ controller: WebController, didUpdateCanGoBack canGoBack: Bool)
+    func webController(_ controller: WebController, didUpdateCanGoForward canGoForward: Bool)
+    func webController(_ controller: WebController, didUpdateEstimatedProgress estimatedProgress: Double)
+    func webController(_ controller: WebController, scrollViewWillBeginDragging scrollView: UIScrollView)
+    func webController(_ controller: WebController, scrollViewDidEndDragging scrollView: UIScrollView)
+    func webController(_ controller: WebController, scrollViewDidScroll scrollView: UIScrollView)
+    func webController(_ controller: WebController, stateDidChange state: BrowserState)
+    func webControllerShouldScrollToTop(_ controller: WebController) -> Bool
+}
+
+
+
+class WebViewController: UIViewController, WebController {
+    weak var delegate: WebControllerDelegate?
+
+    private var browserView = WKWebView()
+    private var progressObserver: NSKeyValueObservation?
+
+    var printFormatter: UIPrintFormatter { return browserView.viewPrintFormatter() }
+    var scrollView: UIScrollView { return browserView.scrollView }
+
+    convenience init() {
+        self.init(nibName: nil, bundle: nil)
+        setupWebview()
+    }
+
+
+    override func loadView() {
+        self.view = browserView
+    }
+//
+//    override func viewPrintFormatter() {
```

Comment:
what's this about

Reply:
Ah, I moved it to a computed property. Good catch!